rfcs/len-metamethod-rawlen.md

```diff
@@ -0,0 +1,43 @@
+# Support `__len` metamethod for tables and `rawlen` function
+
+## Summary
+
+`__len` metamethod will be called by `#` operator on tables, matching Lua 5.2
+
+## Motivation
+
+Lua 5.1 invokes `__len` only on userdata objects, whereas Lua 5.2 extends this to tables. In addition to making `__len` metamethod more uniform and making Luau
```

Comment:
I am sure that Lua 5.1 invokes the `__len` metamethod on *all* data types *except* for tables and strings, not just userdata only as numbers, functions, nil, booleans, and threads can have metatables.
So did I miss something then with the statement "Lua 5.1 invokes `__len` ***only*** on userdata objects" and not including numbers, functions, threads, nil, and booleans?

Should `__len` also invoke on strings as well? (though vanilla Lua never had this and this is likely a feature too esoteric to be added). e.g.
```
function metatable_of_the_strings.__len(self)
    return tonumber(self) or 0
end
print(#"1234") --> 1234
print(#"42") --> 42
print(#"hello world") --> 0
print(rawlen("1234")) --> 4
print(rawlen("42")) --> 2
print(rawlen("hello world")) --> 11
```

Reply:
While you're technically correct in that a host program, or `debug.setmetatable`, can change metatable on other object types, Luau doesn't support `debug.setmetatable` so this is written assuming a user-space program that doesn't use `debug.` API.